src/Model/ContactRelation.php

```diff
@@ -200,4 +206,333 @@ public static function isDiscoverable(string $url, array $contact = [])
+      AND `follows`
+) 
+  AND `id` IN (
+    SELECT `relation-cid`
+    FROM `contact-relation`
+    WHERE `cid` = ?
+      AND `follows`
+)", $sourceId, $targetId]
+		);
+
+		$contacts = DI::dba()->selectToArray(
+			'contact',
+			$condition,
+			[
+				'limit' => [$offset, $count],
+				'order' => [$shuffle ? 'name' : 'RAND()'],
+			]
+		);
+
+		return $contacts;
+	}
+
+
+	/**
+	 * Counts the number of contacts that are followed by both provided public contacts.
+	 *
+	 * @param int   $sourceId  Public contact id
+	 * @param int   $targetId  Public contact id
+	 * @param array $condition Additional condition array on the contact table
+	 * @return int
+	 * @throws Exception
+	 */
+	public static function countCommonFollows(int $sourceId, int $targetId, array $condition = [])
+	{
+		$condition = DBA::mergeConditions($condition,
+			['`id` IN (
+    SELECT `relation-cid`
+    FROM `contact-relation`
+    WHERE `cid` = ?
+      AND `follows`
+) 
+  AND `id` IN (
+    SELECT `relation-cid`
+    FROM `contact-relation`
+    WHERE `cid` = ?
+      AND `follows`
+)', $sourceId, $targetId]
+		);
+
+		$total = DI::dba()->count('contact', $condition);
+
+		return $total;
+	}
+
+	/**
+	 * Returns a paginated list of contacts that are followed by both provided public contacts.
+	 *
+	 * @param int   $sourceId  Public contact id
+	 * @param int   $targetId  Public contact id
+	 * @param array $condition Additional condition array on the contact table
+	 * @param int   $count
+	 * @param int   $offset
+	 * @param bool  $shuffle
+	 * @return array
+	 * @throws Exception
+	 */
+	public static function listCommonFollows(int $sourceId, int $targetId, array $condition = [], int $count = 30, int $offset = 0, bool $shuffle = false)
+	{
+		$condition = DBA::mergeConditions($condition,
+			["`id` IN (
+    SELECT `relation-cid`
+    FROM `contact-relation`
+    WHERE `cid` = ?
+      AND `follows`
+) 
+  AND `id` IN (
+    SELECT `relation-cid`
+    FROM `contact-relation`
+    WHERE `cid` = ?
+      AND `follows`
+)", $sourceId, $targetId]
+		);
+
+		$contacts = DI::dba()->selectToArray(
+			'contact',
+			$condition,
+			[
+				'limit' => [$offset, $count],
+				'order' => [$shuffle ? 'name' : 'RAND()'],
+			]
+		);
+
+		return $contacts;
+	}
+
+	/**
+	 * Counts the number of contacts that follow both provided public contacts.
+	 *
+	 * @param int   $sourceId  Public contact id
+	 * @param int   $targetId  Public contact id
+	 * @param array $condition Additional condition on the contact table
+	 * @return int
+	 * @throws Exception
+	 */
+	public static function countCommonFollowers(int $sourceId, int $targetId, array $condition = [])
+	{
+		$condition = DBA::mergeConditions($condition,
+			['`id` IN (
+    SELECT `cid`
+    FROM `contact-relation`
+    WHERE `relation-cid` = ?
+      AND `follows`
+) 
+  AND `id` IN (
+    SELECT `cid`
+    FROM `contact-relation`
+    WHERE `relation-cid` = ?
+      AND `follows`
+)', $sourceId, $targetId]
+		);
```

Comment:
Some side note: I'm always confused when I see something split in so many rows. I totally lose context and have to collapse the stuff to understand whats happening here.

Also I'm confused about the indention with spaces here.

Reply:
My IDE aligns the `AND` with the `WHERE` with spaces, can you please suggest an indentation that would be better readable for you?

Reply:
Your IDE is bad, take VSCode instead :grimacing: 

Concerning the indentation I guess most irritating is something like the single `)` in line 490. Also I still follow the (ancient) rule of keeping stuff into a single line that fit into a single line without scrolling, especially when it belongs together. (see my PR for this)